scrape/target.go

```diff
@@ -492,3 +503,53 @@ func TargetsFromGroup(tg *targetgroup.Group, cfg *config.ScrapeConfig) ([]*Targe
 	}
 	return targets, failures
 }
+
+// buildTargetLabels builds final target labels based on the given target label set, target group label set and scrape config.
+func buildTargetLabels(targetLSet, targetGroupLSet model.LabelSet, cfg *config.ScrapeConfig) labels.Labels {
+	labelsMap := make(map[string]string, len(targetLSet)+len(targetGroupLSet))
+	// Labels from scrape config.
+	scrapeLabels := []labels.Label{
+		{Name: model.JobLabel, Value: cfg.JobName},
+		{Name: model.ScrapeIntervalLabel, Value: cfg.ScrapeInterval.String()},
+		{Name: model.ScrapeTimeoutLabel, Value: cfg.ScrapeTimeout.String()},
+		{Name: model.MetricsPathLabel, Value: cfg.MetricsPath},
+		{Name: model.SchemeLabel, Value: cfg.Scheme},
+	}
+	for _, l := range scrapeLabels {
+		set(labelsMap, l.Name, l.Value)
+	}
+	// Target group labels.
+	for ln, lv := range targetGroupLSet {
+		set(labelsMap, string(ln), string(lv))
+	}
+
+	// Target labels.
+	for ln, lv := range targetLSet {
+		set(labelsMap, string(ln), string(lv))
+	}
+
+	// Labels from config query params.
+	for k, v := range cfg.Params {
+		if len(v) > 0 {
+			set(labelsMap, model.ParamLabelPrefix+k, v[0])
+		}
+	}
+	return buildLabels(labelsMap)
+}
+
+func buildLabels(ls map[string]string) labels.Labels {
```

Comment:
Sounds like this should be in the label package

Reply:
I didn't want to put this function in the label package since it's a really specific usage to the scrape package, and I didn't want to expose another way to convert the labels.

So are you sure you would like to see this method in the label package ?

Reply:
There is func FromMap(m map[string]string) Labels

Reply:
done @roidelapluie